compiler/rustc_ast/src/token.rs

```diff
@@ -436,6 +436,30 @@ impl Token {
             || self == &OpenDelim(Delimiter::Parenthesis)
     }
 
+    /// Returns `true` if the token can appear at the start of an item.
+    pub fn can_begin_item(&self) -> bool {
+        match self.kind {
+            Ident(name, _) => [
+                kw::Fn,
```

Comment:
This list doesn't exist somewhere else in the parser already, right?

Reply:
No, it doesn't 👍